src/s3_payload.rs

```diff
@@ -0,0 +1,151 @@
+use crate::exports::edgee::protocols::data_collection::Dict;
+use anyhow::anyhow;
+use aws_credential_types::Credentials;
+use aws_sigv4::http_request::{
+    sign, PayloadChecksumKind, SignableBody, SignableRequest, SigningParams, SigningSettings,
+};
+use aws_sigv4::sign::v4;
+use aws_smithy_runtime_api::client::identity::Identity;
+use chrono::offset::Utc;
+use chrono::DateTime;
+use std::collections::HashMap;
+use std::time::SystemTime;
+use uuid::Uuid;
+
+pub struct S3Config {
+    pub access_key: String,
+    pub secret_key: String,
+    pub session_token: String, // could be empty
+    pub region: String,
+    pub bucket: String,
+    pub key_prefix: String, // could be empty
+}
+
+impl S3Config {
+    pub fn new(cred_map: Dict) -> anyhow::Result<Self> {
+        let cred: HashMap<String, String> = cred_map
+            .iter()
+            .map(|(key, value)| (key.to_string(), value.to_string()))
+            .collect();
+
+        let access_key = match cred.get("aws_access_key") {
+            Some(key) => key,
+            None => return Err(anyhow!("Missing AWS Access Key")),
+        }
+        .to_string();
+
+        let secret_key = match cred.get("aws_secret_key") {
+            Some(key) => key,
+            None => return Err(anyhow!("Missing AWS Secret Key")),
+        }
+        .to_string();
+
+        let session_token = match cred.get("aws_session_token") {
+            Some(key) => key,
+            None => "", // session token is optional
+        }
+        .to_string();
```

Comment:
Why not using `Option::unwrap_or_default` since the `Default` for `String` is the empty string
```suggestion
	let session_token = cred.get("aws_session_token").unwrap_or_default();
```

Reply:
nice!

That gives me this error though: `the trait bound `&std::string::String: Default` is not satisfied`

Reply:
This works but it looks odd :) 

```rust
let session_token = cred.get("aws_session_token").unwrap_or(&String::from("")).to_string();
```

Reply:
This looks better to me:

```rust
let session_token = cred.get("aws_session_token").map(String::to_string).unwrap_or_default();
```